3_visualize/src/write_diagnostics.R

```diff
@@ -10,15 +10,23 @@ write_diagnostics <- function(filepath_in, filepath_out){
   eval_data <- readRDS(file.path(filepath_in, 'eval_data'))
   
   # Save the model diagnostics
-  render_data <- list(pgdl_980mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
-                      dl_980mean = filter(eval_data, model_type == 'dl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
-                      pb_980mean = filter(eval_data, model_type == 'pb', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
-                      dl_500mean = filter(eval_data, model_type == 'dl', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
-                      pb_500mean = filter(eval_data, model_type == 'pb', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
-                      dl_100mean = filter(eval_data, model_type == 'dl', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
-                      pb_100mean = filter(eval_data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
-                      pgdl_2mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
-                      pb_2mean = filter(eval_data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
+  render <- function(model, exper){
```

Comment:
😎 😎 😎 😎 😎 

Two things:
1. You could take this one step further and write a second helper function to write list piece:
```r
render_data <- 
    list(pgdl_980mean = render('pgdl', "similar_980"),
         dl_980mean = render('dl',  "similar_980"),
         pb_980mean = render('pb', "similar_980"),
         dl_500mean = render('dl', "similar_500"),
         pb_500mean = render('pb', "similar_500"),
         dl_100mean = render('dl', "similar_100"),
         pb_100mean = render('pb', "similar_100"),
         pgdl_2mean = render('pgdl', "similar_2"),
         pb_2mean = render('pb', "similar_2"))
```
2. move the `render` functions outside of the `write_diagnostics` function to make it clear that you are writing a function to create a table. Then you can call `render` in `write_diagnostics` (as you do in lines 21-29). When R runs `source(3_visualize/src/write_diagnostics.R` it will load both functions so that `write_diagnostics` can access `render`